tests/foreman/ui_airgun/test_contentview.py

```diff
@@ -65,3 +71,43 @@ def test_positive_add_custom_content(session):
         session.contentview.add_yum_repo(cv_name, repo_name)
         cv_values = session.contentview.read(cv_name)
         assert cv_values['yumrepo']['repos']['assigned'][0] == repo_name
+
+
+@tier2
+@upgrade
+def test_positive_end_to_end(session, module_org):
+    """Create content view with yum repo, publish it and promote it to Library
+        +1 env
+
+    :id: 74c1b00d-c582-434f-bf73-588532588d50
+
+    :steps:
+        1. Create Product/repo and Sync it
+        2. Create CV and add created repo in step1
+        3. Publish and promote it to 'Library'
+        4. Promote it to next environment
+
+    :expectedresults: content view is created, updated with repo publish and
+        promoted to next selected env
+
+    :CaseLevel: Integration
+    """
+    repo_name = gen_string('alpha')
+    env_name = gen_string('alpha')
+    cv_name = gen_string('alpha')
+    version = 'Version 1.0'
+    with session:
+        # Create Life-cycle environment
+        session.lifecycleenvironment.create({'name': env_name})
+        # Creates a CV along with product and sync'ed repository
+        create_sync_custom_repo(module_org.id, repo_name=repo_name)
```

Comment:
we can put this step out of the UI session, what do you think?

Reply:
Definitely can, i've already tested that and wanted to submit with the next PR, but yeah, since it's not yet merged better to update this one :)